.github/workflows/github_push.yaml

```diff
@@ -1,68 +1,112 @@
+          - { os: macOS-10.15,  dc: dmd-latest }
+          - { os: windows-2019, dc: dmd-latest }
+
     runs-on: ${{ matrix.os }}
     timeout-minutes: 30
     steps:
+
+    # Checkout this repository and its submodules
     - uses: actions/checkout@v2
       with:
         submodules: true
-        persist-credentials: false
+
+    # Install the D compiler
     - name: Prepare compiler
       uses: mihails-strasuns/setup-dlang@v0.5.0
       with:
           compiler: ${{ matrix.dc }}
           gh_token: ${{ secrets.GITHUB_TOKEN }}
-    - name: Print compiler and dub's version
-      run: |
-        ${DC} --version
-        dub --version
-    - name: Install dependencies
+
+    # Install os-specific packages
+    # Those will show up in the list of steps, but be grayed out,
+    # hence the usage of the `[OSX]` tag
+    - name: '[OSX] Install dependencies & setup environment'
+      if: runner.os == 'macOS'
       run: |
         brew install pkg-config
-    - name: Build & test agora
-      run: |
         ./ci/ci_osx_setup.sh
-        export PATH="${PATH-}:$HOME/bin/"
-        export LIBRARY_PATH="${LD_LIBRARY_PATH-}:/usr/local/lib/"
-        export PKG_CONFIG_PATH="/usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/openssl@1.1/lib/pkgconfig/"
-        ./ci/run.sh
+        echo ::set-env name=LIBRARY_PATH::${LD_LIBRARY_PATH-}:/usr/local/lib/
+        echo ::set-env name=PKG_CONFIG_PATH::/usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/openssl@1.1/lib/pkgconfig/
 
-  linux_build:
-    name: Linux
-    strategy:
-      matrix:
-        os: [ubuntu-18.04]
-        dc: [dmd-master, ldc-master]
-    runs-on: ${{ matrix.os }}
-    timeout-minutes: 30
-    steps:
-    - uses: actions/checkout@v2
-      with:
-        submodules: true
-        persist-credentials: false
-    - name: Prepare compiler
-      uses: mihails-strasuns/setup-dlang@v0.5.0
-      with:
-        compiler: ${{ matrix.dc }}
-        gh_token: ${{ secrets.GITHUB_TOKEN }}
-    - name: Print compiler and dub's version
-      run: |
-        ${DC} --version
-        dub --version
-    - name: Build & test Agora
+    - name: '[Linux] Install dependencies & setup environment'
+      if: runner.os == 'Linux'
       run: |
-        ./ci/ci_linux_setup.sh
-        sudo add-apt-repository ppa:ubuntu-toolchain-r/test
         sudo apt-get update
         sudo apt-get install libsqlite3-dev
         sudo apt-get install libsodium-dev
         sudo apt-get install g++-9
-        ./ci/run.sh
+
+    # Add whatever debugging information can be useful in the long run here
+    - name: Print system informations
+      run: |
+        ${DC} --version
+        dub --version
+
+    # Build and run the tests
+    - name: Build & test Agora
+      run: ./ci/run.sh
+
+    # Documentation: Note that this will push documentation to your fork, too!
+    - name: Build and deploy documentation
+      if: matrix.dc == 'dmd-latest'
```

Comment:
This is likely to be done twice on Mac and Linux now.
As an additional condition, Runner.os == 'Linux' will be required.

Reply:
As mentioned above, the `exclude` takes care of that.